src/histogram.rs

```diff
@@ -710,6 +725,41 @@ impl Drop for LocalHistogram {
     }
 }
 
+/// `LocalHistogramVec` can only be access by a single thread, like `LocalHistogramVec`.
+/// You can use the `LocalHistogramVec`, if you very care the performance.
+pub struct LocalHistogramVec {
+    vec: HistogramVec,
+    local: HashMap<u64, LocalHistogram>,
+}
+
+impl LocalHistogramVec {
+    /// Get a `LocalHistogram` by label values.
+    /// See more [MetricVec::with_label_values]
+    /// (/prometheus/struct.MetricVec.html#method.with_label_values)
+    pub fn with_label_values<'a>(&'a mut self, vals: &[&str]) -> &'a LocalHistogram {
+        let hash = self.vec.v.hash_label_values(vals).unwrap();
+        let vec = &self.vec;
+        self.local
+            .entry(hash)
+            .or_insert_with(|| vec.with_label_values(vals).local())
+    }
+
+    /// Remove a `LocalHistogram` by label values.
+    /// See more [MetricVec::remove_label_values]
+    /// (/prometheus/struct.MetricVec.html#method.remove_label_values)
+    pub fn remove_label_values(&mut self, vals: &[&str]) -> Result<()> {
+        let hash = self.vec.v.hash_label_values(vals)?;
+        self.local.remove(&hash);
+        self.vec.v.delete_label_values(vals)
+    }
+
+    pub fn flush(&mut self) {
```

Comment:
add comment

Reply:
you can use drain to clear all values.

Reply:
I do not think we need drain, drain drops all cached `LocalHistogram`s.